src/prepare_dlc_dev_environment.py

```diff
@@ -560,15 +688,46 @@ def main():
         handle_currency_option(buildspec_paths)
         create_dockerfile_paths(buildspec_paths)
 
+    if override_tags:
+        for buildspec_path in buildspec_paths:
+            override_existing_buildspec(buildspec_path, override_tags)
+
     # handle frameworks to build
     if buildspec_paths:
         overrider.set_test_types(test_types=test_types)
         overrider.set_buildspec(buildspec_paths=buildspec_paths)
 
     LOGGER.info(overrider.overrides)
     write_toml(toml_path, overrides=overrider.overrides)
+
+    changes = {}
+    if restore:
```

Comment:
is there a reason to have duplicate if conditions in the main function? For example, this condition should never be hit by a restore flag, because line 682 returns when the restore option is set